artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java

```diff
@@ -3933,6 +4005,37 @@ void scheduleSlowConsumerReaper(AddressSettings settings) {
       }
    }
 
+   private void enforceRing() {
```

Comment:
could ringsize != -1 check be done here, will allow better escaping and inlining where not being used., e.g. avoids need to call into method below which is many lines long, and unlikely to be inlined

Reply:
(nit)

Reply:
Will do.